File: app/src/main/java/ru/samtakoy/core/screens/log/LogActivity.java
package ru.samtakoy.core.screens.log;

import android.content.Context;
import android.content.Intent;

import androidx.fragment.app.Fragment;

import ru.samtakoy.core.screens.SingleFragmentActivity;

public class LogActivity extends SingleFragmentActivity {


    public static Intent newActivityIntent(Context callerContext){
        return new Intent(callerContext, LogActivity.class);
    }

    @Override
    protected Fragment createFragment() {
        return LogFragment.newFragment();
    }
}

File: androidApp/src/main/kotlin/ru/samtakoy/oldlegacy/core/presentation/log/LogActivity.kt
package ru.samtakoy.oldlegacy.core.presentation.log

import android.content.Context
import android.content.Intent
import androidx.fragment.app.Fragment
import ru.samtakoy.oldlegacy.core.presentation.SingleFragmentActivity
import ru.samtakoy.oldlegacy.core.presentation.log.LogFragment.Companion.newFragment

class LogActivity : SingleFragmentActivity() {
    override fun createFragment(): Fragment {
        return newFragment()
    }

    companion object {
        fun newActivityIntent(callerContext: Context): Intent {
            return Intent(callerContext, LogActivity::class.java)
        }
    }
}
